test/sequential/feeLock.API.feePayment.test.ts

```diff
@@ -0,0 +1,399 @@
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let firstCurrency: BN;
+let feeLockMetadata: any;
+let periodLength: any;
+let feeLockAmount: any;
+let whitelistedTokens: any[];
+const thresholdValue = new BN(30000);
+const defaultCurrencyValue = new BN(10000000);
+const defaultPoolVolumeValue = new BN(1000000);
+
+async function checkErrorSellAsset(
+  reason: string,
+  amount = new BN(1000),
+  buyMgaToken = true
+) {
+  let exception = false;
+  const mangata = await getMangataInstance();
+
+  if (buyMgaToken) {
+    await expect(
+      mangata
+        .sellAsset(
+          testUser1.keyRingPair,
+          firstCurrency.toString(),
+          MGA_ASSET_ID.toString(),
+          amount,
+          new BN(0)
+        )
+        .catch((reason) => {
+          exception = true;
+          throw new Error(reason.data);
+        })
+    ).rejects.toThrow(reason);
+  } else {
+    await expect(
+      mangata
+        .sellAsset(
+          testUser1.keyRingPair,
+          MGA_ASSET_ID.toString(),
+          firstCurrency.toString(),
+          amount,
+          new BN(0)
+        )
+        .catch((reason) => {
+          exception = true;
+          throw new Error(reason.data);
+        })
+    ).rejects.toThrow(reason);
+  }
+  expect(exception).toBeTruthy();
+}
+
+async function clearMgaFromWhitelisted() {
+  let checkMgaWhitelistedToken: boolean;
+  const swapValueThreshold = feeLockMetadata.swapValueThreshold;
+  whitelistedTokens = feeLockMetadata.whitelistedTokens;
+  checkMgaWhitelistedToken = false;
+
+  whitelistedTokens.forEach((element) => {
+    if (element.toString() === MGA_ASSET_ID.toString()) {
+      checkMgaWhitelistedToken = true;
+    }
+  });
+
+  if (checkMgaWhitelistedToken) {
+    const checkEmptyTimeoutConfig = await updateFeeLockMetadata(
+      sudo,
+      new BN(periodLength),
+      new BN(feeLockAmount),
+      swapValueThreshold,
+      [[MGA_ASSET_ID, false]]
+    );
+    await waitSudoOperataionSuccess(checkEmptyTimeoutConfig);
+  }
+}
+
+async function addMgaToWhitelisted() {
+  let checkMgaWhitelistedToken: boolean;
+  const swapValueThreshold = feeLockMetadata.swapValueThreshold;
+  whitelistedTokens = feeLockMetadata.whitelistedTokens;
+  checkMgaWhitelistedToken = false;
+
+  whitelistedTokens.forEach((element) => {
+    if (element.toString() === MGA_ASSET_ID.toString()) {
+      checkMgaWhitelistedToken = true;
+    }
+  });
+
+  if (swapValueThreshold < thresholdValue) {
+    checkMgaWhitelistedToken = false;
+  }
+
+  if (!checkMgaWhitelistedToken) {
+    const updateMgaTimeoutMetadata = await updateFeeLockMetadata(
+      sudo,
+      new BN(periodLength),
+      new BN(feeLockAmount),
+      thresholdValue,
+      [[MGA_ASSET_ID, true]]
+    );
+    await waitSudoOperataionSuccess(updateMgaTimeoutMetadata);
+  }
+}
```

Comment:
What if we do the following with those functions?
* Move them to utils. 
* Instead of reading the status from that global variable ( quite a bad practice ) , we query the chain storage and run the updateFeeLockMetadata if neded.

Reply:
Added `feeLockHelper.ts`, variables were removed